src/app/shared/measur-survey/measur-survey.service.ts

```diff
@@ -0,0 +1,96 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import moment from 'moment';
+import { BehaviorSubject, firstValueFrom } from 'rxjs';
+import { environment } from '../../../environments/environment';
+import { ApplicationInstanceData } from '../../indexedDb/application-instance-db.service';
+import { MeasurUserSurvey } from './experience-survey/experience-survey.component';
+import { DirectoryDbService } from '../../indexedDb/directory-db.service';
+import { Directory } from '../models/directory';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MeasurSurveyService {
+  completedStatus: BehaviorSubject<'sending' | 'success' | 'error'>;
+  showSurveyModal: BehaviorSubject<boolean>;
+  userSurvey: BehaviorSubject<MeasurUserSurvey>;
+
+  constructor(private httpClient: HttpClient, private directoryDbService: DirectoryDbService) {
+    this.showSurveyModal = new BehaviorSubject<boolean>(undefined);
+    this.completedStatus = new BehaviorSubject<'sending' | 'success' | 'error'>(undefined);
+    this.userSurvey = new BehaviorSubject<MeasurUserSurvey>(undefined);
+  }
+
+ async getHasMetUsageRequirements(applicationData: ApplicationInstanceData) {
+      let firstAppInitDate = moment(new Date(applicationData.createdDate));
+      let currentDate = moment(new Date());
+  
+      let hasMetUsageThreshold: boolean;
+      let dateDifference;
+      if (environment.production) {
+        dateDifference = currentDate.diff(firstAppInitDate, 'days');
+        hasMetUsageThreshold = dateDifference >= 30 || applicationData.appOpenCount >= 10;
+      } else {
+        dateDifference = currentDate.diff(firstAppInitDate, 'seconds');
+        hasMetUsageThreshold = dateDifference >= 120 || applicationData.appOpenCount >= 2;
+      }
+      
+      return hasMetUsageThreshold;
+  }
+
+  /**
+   * Check if is legacy user and has used app for 30 days 
+   */
+  async checkIsExistingUser() {
+    let currentDate = moment(new Date());
+    let allDirs: Directory[] = await firstValueFrom(this.directoryDbService.getAllDirectories());
+    let topLevelDirInitDate = allDirs.find(dir => dir.parentDirectoryId === null && dir.name === 'All Assessments')?.createdDate; 
```

Comment:
This should be robust enough. Users can't change All Assessments dir name or add a parentDirectoryId unless they hack json